cubeviz/controls/flux_unit_registry.py

```diff
@@ -0,0 +1,289 @@
+import os
+import yaml
+
+from astropy import units as u
+
+DEFAULT_FLUX_UNITS_CONFIGS = os.path.join(os.path.dirname(__file__), 'registered_flux_units.yaml')
+
+
+def _add_unit_to_list(unit_list, target_unit):
+    """
+    Given a list of units, add target units
+    after checking for duplicates
+    :param unit_list: list of units or string
+    :param target_unit: unit or string
+    :return: updated unit
+    """
+
+    if isinstance(target_unit, str):
+        current_unit = u.Unit(target_unit)
+        string_unit = target_unit
+    else:
+        current_unit = target_unit
+        string_unit = target_unit.to_string()
+
+    duplicate = False
+    for unit in unit_list:
+        if isinstance(unit, str):
+            unit = u.Unit(unit)
+        if unit == current_unit:
+            duplicate = True
+            break
+    if not duplicate:
+        unit_list.append(string_unit)
+    return unit_list
+
+
+class FluxUnitRegistry:
+    """
+    Saves a list of spectral flux density units.
+    """
+    def __init__(self):
+        self._model_unit = u.Jy  # Will be used for compatibility checks
+        self.runtime_defined_units = []  # Stores list of runtime units
+
+    @staticmethod
+    def _locally_defined_units():
+        """
+        list of defined spectral flux density units.
+        :return: list of unit strings
+        """
+        units = ['Jy', 'mJy', 'uJy',
+                 'W / (m2 Hz)',
+                 'eV / (s m2 Hz)',
+                 'erg / (s cm2)',
+                 'erg / (s cm2 um)',
+                 'erg / (s cm2 Angstrom)',
+                 'erg / (s cm2 Hz)',
+                 'ph / (s cm2 um)',
+                 'ph / (s cm2 Angstrom)',
+                 'ph / (s cm2 Hz)']
+        return units
+
+    def is_compatible(self, unit):
+        """
+        Check if unit can be converted to units
+        in this registry. Uses model unit to check.
+        :param unit: Unit or unit str
+        :return: bool
+        """
+        try:
+            self._model_unit.to(unit, equivalencies=u.spectral_density(3500 * u.AA))
+            return True
+        except u.UnitConversionError:
+            return False
+
+    def get_unit_list(self, current_unit=None):
```

Comment:
I am a little confused. This function seems to both add a unit to the list AND get the list. If that is the case, I think it would be better to have these as separate functions as that would probably be cleaner.  But, maybe I am not understanding.

Reply:
This is correct. The call to this function actually constructs a list returns a list unique to current unit. 
I could refactor this but it would basically be a redundent function that calls this one.